src/env/log_fd/plain.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2017-present, PingCAP, Inc. Licensed under Apache-2.0.
+
+use crate::env::{Handle, Permission};
+
+use fail::fail_point;
+use parking_lot::RwLock;
+
+use std::fs::{File, OpenOptions};
+use std::io::{Error, ErrorKind, Read, Result, Seek, SeekFrom, Write};
+use std::path::Path;
+use std::sync::Arc;
+
+pub struct LogFd(Arc<RwLock<File>>);
```

Comment:
This implementation kind of defeats the point of `Handle` abstraction, which is to be efficiently shared between multiple readers/writers. I like seeing a fallback solution that covers all platforms other than Windows, but this implementation is too inefficient for production use.

I can think of two ways to go around this:

(1) use raw windows HANDLE here like in https://github.com/tikv/raft-engine/pull/69, it has the downside of windows-only.
(2) further separate `LogFile` into unix and windows version. The windows version looks something like below. It has the downside of needing to port lots of failpoints again.
```rust
struct LogFd(File);
struct LogFile(File);
impl LogFd {
  fn new_writer(&self) -> LogFile {
    LogFile(self.0.try_clone().unwrap())
  }
}
```